app/assets/javascripts/angular/components/organizationCleanup/organizationCleanup.component.js

```diff
@@ -21,10 +22,73 @@ function organizationCleanupController(
     this.archiveBeforeDate = new Date();
     this.archiveInactivityDate = new Date();
     this.checkIcon = checkIcon;
+    this.warningIcon = warningIcon;
+
+    // Opens up modal depending on the type
+    const confirmModal = async (type, archiveDate) => {
+        switch (type) {
+            case 'archive_by_inactivity': {
+                let archiveInactive = date => {
+                    archiveContacts('leaders_last_sign_in_at')(date);
+                };
+
+                const InactiveModal = $uibModal.open({
+                    component: 'iconModal',
+                    resolve: {
+                        title: () =>
+                            $filter('t')(
+                                `Are you sure you want to archive these contacts by inactivity? `,
+                            ),
+                        closeLabel: () => $filter('t')('Ok'),
+                        dismissLabel: () => $filter('t')('Cancel'),
+                    },
+                    windowClass: 'pivot_theme',
+                    backdrop: 'static',
+                    keyboard: false,
+                });
+
+                try {
+                    const InactiveResponse = await InactiveModal.result;
+                    archiveInactive(archiveDate);
+                } catch (err) {
+                    return null;
+                }
+
+                break;
+            }
+            case 'archive_by_date': {
+                let archiveBefore = date => {
+                    archiveContacts('persons_inactive_since')(date);
+                };
+                const DateModal = $uibModal.open({
+                    component: 'iconModal',
+                    resolve: {
+                        title: () =>
+                            $filter('t')(
+                                `Are you sure you want to archive these contacts by date?`,
+                            ),
+                        closeLabel: () => $filter('t')('Ok'),
+                        dismissLabel: () => $filter('t')('Cancel'),
+                    },
+                    windowClass: 'pivot_theme',
+                    backdrop: 'static',
+                    keyboard: false,
+                });
+                try {
+                    const DateResponse = await DateModal.result;
+                    archiveBefore(archiveDate);
+                } catch (error) {
+                    return null;
+                }
+                break;
+            }
+            default:
+                return null;
```

Comment:
You could probably just return undefined here and above unless something is expecting null
```suggestion
                return;
```

Reply:
For some reason if I have just return, prettier is getting rid of it? Is just having return undefined; okay or should I look into trying to turn off this prettier formatting option?